### PR TITLE
nix: machine -> machine-generic, add machine-cm3

### PR DESCRIPTION
**Pull request by flokli**
Add a custom target for the CM3 image, which requires a custom bootloader, and a hardcoded DTB name.

### CODE REVIEW

File: nix/configuration-cm3.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    cp ${pkgs.raspberrypifw}/share/raspberrypi/boot/bcm2710-rpi-cm3.dtb firmware/
+  '';
+
+  # u-boot doesn't know the proper device tree name, so cannot pick from FTDIR
+  # on its own.
+  hardware.deviceTree.name = "broadcom/bcm2837-rpi-cm3-io3.dtb";
+
+  # The sdcard seems to be a bit slow to come up.
+  # Switch to systemd-initrd, hoping it's a bit more permissive.
+  boot.initrd.systemd.enable = true;
```
Comment (flokli) [inline nix/configuration-cm3.nix]: This could generally be used in the generic config, will update.

Comment (flokli) [reply nix/configuration-cm3.nix]: Done.

**Comment by freyacodes** [comment]
Is this ready to be merged? I heard that you had issues with HDMI on the compute modules

**Comment by flokli** [comment]
Yes, this is ready for merge regardless. We need the FDT we configure here, we need a separate target, …

There might be followup fixes necessary, but they can live in followup PRs.

**Review by freyacodes — APPROVED** [review]

**Pull request merged**